lib/puppet/provider/package/windows/msi_package.rb

```diff
@@ -23,7 +23,7 @@ def self.from_registry(name, values)
           MsiPackage.new(get_display_name(values),
                          values['DisplayVersion'],
                          name, # productcode
-                         inst.ProductInfo(name, 'PackageCode'))
+                         inst.ProductInfo(name, 'PackageCode').encode(Encoding::UTF_8))
```

Comment:
The fortunate thing here is that this a GUID, so shouldn't ever be outside of the range of `[0-9A-F\-]`

That said, Ruby is still going to pass this out as a current codepage string (based on `Encoding.default_external`), which can make Ruby concatenation (and other string ops) explode depending on where they occur.

The string smashing was the root cause of https://github.com/puppetlabs/puppet/pull/5113